src/components/Results.tsx

```diff
@@ -70,11 +71,16 @@ export function Results() {
         })
         .then(function (response) {
           console.log(response);
-          setResults(normalizeResults(response.data));
+          if (response.status === 200) {
+            setResults(normalizeResults(response.data));
+          } else {
+            setError(response.status);
+          }
           setIsLoading(false);
         })
         .catch(function (error) {
           console.log(error);
+          setError(999);
           setIsLoading(false);
         });
```

Comment:
You should handle the error in the `catch` side, you will always run that function if it's a 5xx or a 4xx. So I would do:

```suggestion
.then(function (response) {
  setResults(normalizeResults(response.data));
  setIsLoading(false);
})
.catch(function (error) {
  setError(error.status);
  setIsLoading(false);
});
```

Reply:
malditos robots